Mobile/lib/Components/TaskList.dart

```diff
@@ -43,23 +46,46 @@ class _TasklistState extends State<Tasklist> {
     _getTasks();
   }
 
-  Future<Map<String, dynamic>> _getTasksFromServer() async {
+  Future<http.Response> _getTasksFromServer() async {
+    Api api = Api();
     int familyId = appState.family!.id;
+    String? jwt = await appState.storage.read(key: 'auth_token');
     return switch (widget.listType) {
-      TasklistType.All => await appState.getTasks('/all/$familyId'),
-      TasklistType.Available => await appState.getTasks('/available/$familyId'),
-      TasklistType.Assigned => await appState.getTasks('/assigned/$familyId'),
-      TasklistType.Completed => await appState.getTasks('/completed/$familyId'),
-      TasklistType.Pending => await appState.getTasks('/pending/$familyId'),
+      TasklistType.All => await api.getTasks('all/$familyId', jwt),
+      TasklistType.Available => await api.getTasks('available/$familyId', jwt),
+      TasklistType.Assigned => await api.getTasks('assigned/$familyId', jwt),
+      TasklistType.Completed => await api.getTasks('completed/$familyId', jwt),
+      TasklistType.Pending => await api.getTasks('pending/$familyId', jwt),
     };
   }
 
   void _getTasks() async {
-    Map<String, dynamic> response = await _getTasksFromServer();
-    if (response['statusCode'] == 200) {
-      appState.addListOfTasks(response['tasks'], widget.listType);
+    http.Response response = await _getTasksFromServer();
+    var jsonData = json.decode(response.body);
+    if (response.statusCode == 200) {
+      List<Task> newTasks = [];
+      if (jsonData.isEmpty) {
+        appState.addListOfTasks(newTasks, widget.listType);
+      }
```

Comment:
Seems redundant.
Hvis json data er tomt så bliver appstate add list of tasks kalt 2 gange både her før for loop samt lige efter for loop.

```suggestion
```